### PR TITLE
PEP 440 version check in CI 

### LINKED ISSUE #831: Dev version number in wrong format
Whenever we compile _tskitmodule or make a package, we get:

```
/home/jk/.local/lib/python3.8/site-packages/setuptools/dist.py:458: UserWarning: Normalizing '0.3.1dev1' to '0.3.1.dev1'
  warnings.warn(tmpl.format(**locals()))
```

It's very easy to get this wrong, so maybe we should add something to CI to validate the version number?

### PR DESCRIPTION
**Pull request by benjeffery**
Fixes #831 

I've left the incorrect version number to show this works.

### CODE REVIEW

File: .circleci/config.yml [hunk 1]
```diff
@@ -43,6 +45,10 @@ jobs:
           command: |
             pre-commit run --all-files --show-diff-on-failure
 
+      - run:
+          name: Check python version is PEP 440 compliant
+          command: |
+            cd python && python -c "from packaging.version import Version;from tskit import _version; assert(str(Version(_version.tskit_version))==_version.tskit_version)"
```
Comment (jeromekelleher) [inline .circleci/config.yml:51]: Looks good. I was imaging we just put it in as a standard unit test, though, and we can add ``packaging`` as a dev requirement. Would this not work?

Comment (benjeffery) [reply .circleci/config.yml]: Yep that would work too, and I guess is more portable. I've updated the branch with the check done as a unit test.

**Comment by jeromekelleher** [comment]
Excellent, this should keep us out of trouble. I guess we should fix the version number now and merge before we make hte bugfix release?

**Comment by benjeffery** [comment]
I've corrected the number - let's merge now and do a separate "release prep" PR to fix it to `0.3.1`?

**Review by jeromekelleher — APPROVED** [review]

**Comment by jeromekelleher** [comment]
Currently failing for windows - need a add ``packaging`` to the list of packages we're installing on appveyor manually (although we should really just bundle these into a CI/appveyor-pypi.txt file or something).

**Pull request merged**